compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64ZeroMemoryOp.java

```diff
@@ -180,8 +180,11 @@ protected void emitCode(CompilationResultBuilder crb, AArch64MacroAssembler masm
                     masm.add(64, size, size, 8);
                 }
             } else {
-                masm.cmp(64, size, 8);
-                masm.branchConditionally(ConditionFlag.LT, tail);
+                if (isUnaligned) {
+                    // After aligning base, we may have size less than 8. Need to check again.
+                    masm.cmp(64, size, 8);
+                    masm.branchConditionally(ConditionFlag.LT, tail);
```

Comment:
The condition should be LE, I think.

In this else block, if "size" here is exactly 8, and base is not 16-byte aligned, we firstly store 8 bytes and subtract the size by 8. Then the size becomes 0, and the following code would do a STP (at line 198) without a check.

Please double check this again.

Reply:
Good catch! This is for size<8 and we then perform per-byte zeroing. I think the actual fix to add `masm.jmp(mainCheck);` after storing 8 byte, i.e. L192.